sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/ColumnVector.java

```diff
@@ -518,19 +519,13 @@ private void throwUnsupportedException(int requiredCapacity, Throwable cause) {
   public abstract double getDouble(int rowId);
 
   /**
-   * Puts a byte array that already exists in this column.
-   */
-  public abstract void putArray(int rowId, int offset, int length);
-
-  /**
-   * Returns the length of the array at rowid.
+   * After writing array elements to the child column vector, call this method to set the offset and
+   * length of the written array.
    */
-  public abstract int getArrayLength(int rowId);
-
-  /**
-   * Returns the offset of the array at rowid.
-   */
-  public abstract int getArrayOffset(int rowId);
+  public void arrayWriteEnd(int rowId, int offset, int length) {
```

Comment:
no there is no `arrayWriteStart`.

Maybe I should pick another name, how about `putArrayOffsetAndLength` or just keep the original name?